### PR TITLE
Add new diagnostic plot scripts for SCDCalibratePanels2 output

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

- Original Gitlab user story: [SCD302](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/302)
- This is #31534 into `master`

This PR introduce a new plotting script that takes in a peaks workspace to generate a correlation plot of the observed detector ID v.s. calculated detector ID (row and col).
The plot are separated on a per bank base.

**To test:**
- First, load the attached testing data ([SC100K_Monoclinic_C.zip](https://github.com/mantidproject/mantid/files/6517203/SC100K_Monoclinic_C.zip)) with the following code

```python
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

LoadIsawPeaks('SC100K_Monoclinic_C.integrate', OutputWorkspace='pws')
LoadIsawUB('pws', 'SC100K_Monoclinic_C.mat')
```
- Second, import and generate the plots with the following code
```python
from SCD_Reduction.SCDCalibratePanels2Diagnostics import SCDCalibratePanels2DiagnosticsPlot

config = {
    "prefix": "idealTOPAZ",
    "type": "png",
    "saveto": "./tmp",
    "mode": "modern",  # slow, but prettier
#     "mode": "legacy",    # fast, and concise
    }
pws = mtd['pws']

SCDCalibratePanels2DiagnosticsPlot(
    pws,
    ["bank13", "bank14", "bank16"],
    config,
    showPlots=True,
    )
```
You should be getting plots similar to the one showing here

- legacy style
idealTOPAZ_bank13
- modern style
idealTOPAZ_bank13

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by zjmorgan** [comment]
Getting `AttributeError: 'Figure' object has no attribute 'add_gridspec'` when trying to run the example. I wonder if this is a Matplotlib version problem?

**Comment by KedoKudo** [comment]
> Getting `AttributeError: 'Figure' object has no attribute 'add_gridspec'` when trying to run the example. I wonder if this is a Matplotlib version problem?

I believe it is a version issue.  You might need to update your matplotlib version or switch back to legacy format, which does not use the grid layout.

**Review by zjmorgan — COMMENTED** [review]
> > Getting `AttributeError: 'Figure' object has no attribute 'add_gridspec'` when trying to run the example. I wonder if this is a Matplotlib version problem?
> 
> I believe it is a version issue. You might need to update your matplotlib version or switch back to legacy format, which does not use the grid layout.

That latter works!

File: scripts/SCD_Reduction/SCDCalibratePanels2Diagnostics.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+    # iterative through peaks to generate data
+    for i in range(npeaks):
+        entry = pws.row(i)
+        rowsID_observed[i] = entry["Row"]
+        colsID_observed[i] = entry["Col"]
+        # compute the calculated entry now
+        pk = pws.getPeak(i)
+        gm = pk.getGoniometerMatrix()
+        hkl = np.array(pk.getHKL())
```
[the remark this answers is not in the document]
Comment (KedoKudo) [reply scripts/SCD_Reduction/SCDCalibratePanels2Diagnostics.py:185]: okay, I will make the change.

**Comment by zjmorgan** [comment]
Code looks good and provides similar diagnostics to ISAW. Changing this to `IntHKL` should satisfy all requirements.

**Review by zjmorgan — CHANGES_REQUESTED** [review]
Change 'getHKL' to 'getIntHKL' to compare the integer indices.

**Review by zjmorgan — APPROVED** [review]

**Pull request merged**